graspologic/utils/utils.py

```diff
@@ -157,19 +159,71 @@ def import_edgelist(
 
 
 def is_symmetric(X):
-    return np.array_equal(X, X.T)
+    return abs(X - X.T).sum() == 0
 
 
 def is_loopless(X):
     return not np.any(np.diag(X) != 0)
 
 
-def is_unweighted(X):
-    return ((X == 0) | (X == 1)).all()
+def is_unweighted(
+    graph: Union[
+        np.ndarray,
+        scipy.sparse.csr.csr_matrix,
+        nx.Graph,
+        nx.DiGraph,
+        nx.MultiGraph,
+        nx.MultiDiGraph,
+    ],
+    weight_attribute: Any = "weight",
+):
+    """
+    Attempts to determine if the provided graph is weighted.
+
+    Parameters
+    ----------
+    graph : Union[np.ndarray, scipy.sparse.csr.csr_matrix, nx.Graph, nx.DiGraph, nx.MultiGraph, nx.MultiDigraph]
+        The graph to test for weightedness. If a networkx graph, we can just ask it directly by querying the weight
+        attribute specified on every edge. It's possible an individual edge can be weighted but the full graph is not.
+        If an adjacency matrix defined by a numpy.ndarray or scipy.sparse.csr.csr_matrix, we check every value; if
+        they are only 0 and 1, we claim the graph is unweighted.
+    weight_attribute : Any
+        Default is ``weight``. Only used for networkx, and used on the edge data dictionary as a key to look up the
+        weight.
+
+    Returns
+    -------
+    bool
+        True if unweighted, False if weighted
+
+    Raises
+    ------
+    TypeError
+        If the provided graph is not a numpy.ndarray, scipy.sparse.csr.csr_matrix, or nx.Graph
+    """
+    if isinstance(graph, np.ndarray):
+        return ((graph == 0) | (graph == 1)).all()
+    elif isinstance(graph, scipy.sparse.csr.csr_matrix):
+        # brute force.  if anyone has a better way, please PR
+        rows, columns = graph.nonzero()
+        for i in range(0, len(rows)):
+            if graph[rows[i], columns[i]] != 1 or graph[rows[i], columns[i]] != 0:
+                return False
+        return True
+    elif isinstance(graph, nx.Graph):
+        return nx.is_weighted(graph, weight=weight_attribute)
+    else:
+        raise TypeError(
+            "This function only works on numpy.ndarray or scipy.sparse.csr.csr_matrix instances"
+        )
 
 
 def is_almost_symmetric(X, atol=1e-15):
-    return np.allclose(X, X.T, atol=atol)
+    # if isspmatrix(X):
+    #     return abs(X - X.T).max() <= atol
+    # else:
+    #     return np.allclose(X, X.T, atol=atol)
```

Comment:
```suggestion
```

Reply:
I'll incorporate this iff we do the same in #523 - I am trying to minimize the chance of there being a textual merge conflict when both branches come in.  That said, I totally agree that we should!